soknad-felles/src/main/java/no/nav/k9/søknad/felles/SelvstendigNæringsdrivende.java

```diff
@@ -0,0 +1,212 @@
+package no.nav.k9.søknad.felles;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.Valid;
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.unmodifiableMap;
+import static no.nav.k9.søknad.felles.Periode.Utils.leggTilPeriode;
+import static no.nav.k9.søknad.felles.Periode.Utils.leggTilPerioder;
+
+public class SelvstendigNæringsdrivende {
+
+    @JsonProperty("perioder")
+    @Valid
+    @NotNull
+    @NotEmpty
+    public final Map<Periode, SelvstendigNæringsdrivendePeriodeInfo> perioder;
+
+    @JsonProperty("organisasjonsnummer")
+    @NotNull
+    public final Organisasjonsnummer organisasjonsnummer;
+
+    public static SelvstendigNæringsdrivende.Builder builder() {
+        return new SelvstendigNæringsdrivende.Builder();
+    }
+
+    @JsonCreator
+    public SelvstendigNæringsdrivende(
+            @JsonProperty("perioder") Map<Periode, SelvstendigNæringsdrivendePeriodeInfo> perioder,
+            @JsonProperty("organisasjonsnummer") Organisasjonsnummer organisasjonsnummer
+    ) {
+        this.perioder = (perioder == null) ? emptyMap() : unmodifiableMap(perioder);
+        this.organisasjonsnummer = organisasjonsnummer;
+    }
+
+    public static final class Builder {
```

Comment:
trengs builder?

Reply:
Samme svar som for frilanser.
https://github.com/navikt/k9-format/pull/41#discussion_r410021430